misc/jenkins/checkbinaryfiles.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Check binary files that might be wrongly committed.
+#
+# Usage: checkbinaryfiles.sh from to
+
+set -e
+
+from=${1:-origin/master}
+to=${2:-HEAD}
+
+err=0
+
+cd `git rev-parse --show-toplevel`
+
+err=0
+
+echo "checklicense $from (`git rev-parse "$from"`) .. $to (`git rev-parse "$to"`)"
+git diff --name-only $from..$to -z > checkbinaryfiles_files.out
+if test \! -s checkbinaryfiles_files.out; then
+	echo "No files files"
+	exit 0
+fi
+xargs -0 file --mime < checkbinaryfiles_files.out | grep application > checkbinaryfiles_application.out
```

Comment:
a rule of thumb that encloses unwanted things like: application/x-executable application/x-sharedlib application/octet-stream etc.